src/test/java/pages/MainPage.java

```diff
@@ -68,7 +103,96 @@ public String getLoginViaNickNameMessage() {
      * Display message "Cart" while entering the shopping cart
      * @return return message of shoppingCartMessage element
      */
-    public String getShoppingCartMessage() {
-        return shoppingCartMessage.getText().toLowerCase().trim();
+    public String getLoginWarningMessage() {
+        WaitUtils.waitForVisibility(incorrectCredentialsWarning);
+        return incorrectCredentialsWarning.getText().toLowerCase().trim();
+    }
+
+    /***
+     * While entering no password, a warning message appears
+     * @return warning message
+     */
+    public String getPasswordInputWarning() {
+        WaitUtils.waitForVisibility(emptyPasswordWarning);
+        return emptyPasswordWarning.getText().toLowerCase().trim();
+    }
+
+    /***
+     * While entering a phrase into the input field, a hint appears
+     * @return hint message
+     */
+    public String getSearchInputHint() {
+        driver.switchTo().frame(driver.findElement(By.className("modal-iframe")));
```

Comment:
why do you use driver.findElement() here?

Reply:
fixed